src/parser/ParseNode.mts

```diff
@@ -0,0 +1,2328 @@
+  export interface MultiplicativeExpression extends BaseParseNode {
+    readonly type: 'MultiplicativeExpression';
+    readonly MultiplicativeExpression: MultiplicativeExpressionOrHigher;
+    readonly MultiplicativeOperator: MultiplicativeOperator;
+    readonly ExponentiationExpression: ExponentiationExpressionOrHigher;
+  }
+
+  // MultiplicativeOperator : one of
+  //   `*`  `/`  `%`;
+  export type MultiplicativeOperator = '*' | '/' | '%';
+
+  // AdditiveExpression :
+  //   MultiplicativeExpression
+  //   AdditiveExpression `+` MultiplicativeExpression
+  //   AdditiveExpression `-` MultiplicativeExpression
+  export type AdditiveExpressionOrHigher =
+    | MultiplicativeExpressionOrHigher
+    | AdditiveExpression;
+
+  // AdditiveExpression (partial) :
+  //   AdditiveExpression `+` MultiplicativeExpression
+  //   AdditiveExpression `-` MultiplicativeExpression
+  export interface AdditiveExpression extends BaseParseNode {
+    readonly type: 'AdditiveExpression';
+    readonly operator: '+' | '-';
+    readonly AdditiveExpression: AdditiveExpressionOrHigher;
+    readonly MultiplicativeExpression: MultiplicativeExpressionOrHigher;
+  }
+
+  // ShiftExpression :
+  //   AdditiveExpression
+  //   ShiftExpression `<<` AdditiveExpression
+  //   ShiftExpression `>>` AdditiveExpression
+  //   ShiftExpression `>>>` AdditiveExpression
+  export type ShiftExpressionOrHigher =
+    | AdditiveExpressionOrHigher
+    | ShiftExpression;
+
+  // ShiftExpression (partial) :
+  //   ShiftExpression `<<` AdditiveExpression
+  //   ShiftExpression `>>` AdditiveExpression
+  //   ShiftExpression `>>>` AdditiveExpression
+  export interface ShiftExpression extends BaseParseNode {
+    readonly type: 'ShiftExpression';
+    readonly operator: '<<' | '>>' | '>>>';
+    readonly ShiftExpression: ShiftExpressionOrHigher;
+    readonly AdditiveExpression: AdditiveExpressionOrHigher;
+  }
+
+  // RelationalExpression :
+  //   ShiftExpression
+  //   RelationalExpression `<` ShiftExpression
+  //   RelationalExpression `>` ShiftExpression
+  //   RelationalExpression `<=` ShiftExpression
+  //   RelationalExpression `>=` ShiftExpression
+  //   RelationalExpression `instanceof` ShiftExpression
+  //   RelationalExpression `in` ShiftExpression
+  export type RelationalExpressionOrHigher =
+    | ShiftExpressionOrHigher
+    | RelationalExpression;
+
+  // RelationalExpression (partial) :
+  //   RelationalExpression `<` ShiftExpression
+  //   RelationalExpression `>` ShiftExpression
+  //   RelationalExpression `<=` ShiftExpression
+  //   RelationalExpression `>=` ShiftExpression
+  //   RelationalExpression `instanceof` ShiftExpression
+  //   RelationalExpression `in` ShiftExpression
+  export interface RelationalExpression extends BaseParseNode {
+    readonly type: 'RelationalExpression';
+    readonly operator: '<' | '>' | '<=' | '>=' | 'instanceof' | 'in';
+    readonly PrivateIdentifier?: PrivateIdentifier;
```

Comment:
`#x in y`

Reply:
is PrivateIdentifier a RelationalExpression? according to the production, it looks like it should be in the `readonly RelationalExpression?: RelationalExpressionOrHigher;`

Reply:
No, it isn't. PrivateIdentifier is only allowed on its own when on the left side of an `in` expression, so it can't be a RelationalExpression on its own. Also, this type is reflecting the current behavior of the parser.

Reply:
oh I understand. Your comment is missing the [PrivateIdentifier](https://tc39.es/ecma262/multipage/ecmascript-language-lexical-grammar.html#prod-PrivateIdentifier) in [ShiftExpression](https://tc39.es/ecma262/multipage/ecmascript-language-expressions.html#prod-ShiftExpression) production. I'll add it for you